codice/ddf
# DDF-4639 Update strategy for UI extensions

#### What does this PR do?
Updates strategy for UI extensions to leverage javascript / typescript.

This also involved pulling out 3 sub packages, now located here:
https://github.com/connexta/ace
https://github.com/connexta/eslint-plugin
https://github.com/connexta/icons

Also fixes an issue in https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/component/filter-builder/filter-serialization.js#L59 that was causing near filters to fail.

#### Who is reviewing it? 
@Bdthomson 
@djblue 
@adimka 
@rzwiefel 

#### How should this be tested?
Verify that the navigator and the routes work as expected.  Also, if you have access to a downstream project.  Verify that these changes don't break things there.  I have PRs in respective downstream projects to make them compatible with this PR.

#### What are the relevant tickets?
Fixes: #4639
